# Exploratory Data Analysis (EDA)

Nesta etapa do projeto, entramos na fase de **Exploratory Data Analysis (EDA)** dentro da metodologia **CRISP-DM**.

## Onde estamos no CRISP-DM

Até aqui, as seguintes fases já foram concluídas:

- **Business Understanding**  
  Definição do problema, contexto público dos dados e objetivos analíticos.
- **Data Understanding**  
  Inspeção inicial dos dados brutos, análise de estrutura, schemas, tipos de dados,
  valores ausentes e qualidade geral.
- **Data Preparation**  
  Limpeza, padronização, normalização, tratamento de colunas, criação de variáveis
  auxiliares e geração do dataset final para análise.

Agora, iniciamos a fase de **Exploratory Data Analysis (EDA)**.

### Objetivos da fase de EDA

Os principais objetivos desta fase são:

- Explorar estatisticamente o dataset preparado (`data/processed`)
- Identificar padrões, tendências e distribuições relevantes
- Analisar relações entre variáveis (ex.: cargos, salários, tempo de serviço)
- Detectar possíveis outliers ou comportamentos atípicos
- Gerar hipóteses analíticas que poderão ser aprofundadas posteriormente
- Produzir insights compreensíveis tanto para:
  - Público geral (cidadãos)
  - Recrutadores e profissionais da área de dados

### Papel da EDA neste projeto

A EDA é a ponte entre os **dados tratados** e a **comunicação dos resultados**.

Os achados desta etapa servirão de base para:

- Construção de narrativas analíticas
- Visualizações interativas
- Publicação de resultados em dashboards (ex.: Streamlit)
- Demonstração de habilidades analíticas em portfólio profissional

Nenhuma modelagem preditiva será realizada neste momento.  
O foco está em **compreender profundamente os dados** antes de qualquer etapa avançada.


In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [6]:
# Definindo quantas colunas o pandas irá exibir
pd.set_option("display.max_columns", 100)

# Definindo como números float serão exibidos (2 casas decimais)
pd.set_option("display.float_format", lambda x: f"{x:,.2f}")

In [7]:
df_serv = pd.read_csv("../data/processed/servidores_municipais_prepared.csv", sep=",", encoding="utf-8")

In [11]:
df_serv.head()

id_servidor sexo  \
0  fd384acafe2d04b4525169770984b83246828743f0e8aa...    M   
1  379ea2db2ce8b452ca96cbd1ee182fbda32dd884195931...    M   
2  379ea2db2ce8b452ca96cbd1ee182fbda32dd884195931...    M   
3  79d7249fd3bb08e95ddf2596765773e22927a0113cf95d...    M   
4  79d7249fd3bb08e95ddf2596765773e22927a0113cf95d...    M   

                           cargo categoria_cargo    tipo_pagamento  proventos  \
0  AJUDANTE DE SERVICOS DIVERSOS     operacional  vale_alimentacao   1,000.00   
1                      MOTORISTA     operacional  vale_alimentacao   1,000.00   
2                      MOTORISTA     operacional      folha_mensal   3,675.13   
3         OFICIAL ADMINISTRATIVO  administrativo      folha_mensal   6,257.37   
4         OFICIAL ADMINISTRATIVO  administrativo  vale_alimentacao   1,000.00   

   descontos  liquido  carga_horaria_semanal data_admissao data_desligamento  \
0       0.00 1,000.00                     40    2006-06-19             ATIVO   
1       0.00 1,000.00                     40    2012-08-13             ATIVO   
2     412.48 3,262.65                     40    2012-08-13             ATIVO   
3   1,459.90 4,797.47                     35    2006-03-20             ATIVO   
4       0.00 1,000.00                     35    2006-03-20             ATIVO   

   mes  
0  jan  
1  jan  
2  jan  
3  jan  
4  jan

In [14]:
df_serv.tail()

id_servidor sexo  \
23992  f061d46cb3dc7c55a5d2d891d6d6105644bc10d7b301e8...    F   
23993  3e5ceb7467aab950efa7a34a6a8806606a3904629ad95b...    F   
23994  3e5ceb7467aab950efa7a34a6a8806606a3904629ad95b...    F   
23995  10a60cdde464dbcccbef4f84fb09db4e4a5642708caeae...    F   
23996  3c40f1de7d5963c46fc43919f6783f5c21c5c633718e2e...    F   

                                        cargo categoria_cargo  \
23992           AJUDANTE DE SERVICOS DIVERSOS     operacional   
23993           AJUDANTE DE SERVICOS DIVERSOS     operacional   
23994           AJUDANTE DE SERVICOS DIVERSOS     operacional   
23995                              MERENDEIRA        educacao   
23996  PROFESSOR DE EDUCAÇÃO BÁSICA I - PEB I        educacao   

               tipo_pagamento  proventos  descontos  liquido  \
23992            folha_mensal   2,299.65     321.95 1,977.70   
23993  Fechamento 13º Salário   2,298.25   1,132.26 1,165.99   
23994        vale_alimentacao   1,000.00       0.00 1,000.00   
23995            folha_mensal   3,109.45     369.42 2,740.03   
23996  Fechamento 13º Salário   4,483.67   3,061.64 1,422.03   

       carga_horaria_semanal data_admissao data_desligamento  mes  
23992                     40    2013-04-01             ATIVO  dez  
23993                     44    2009-12-21             ATIVO  dez  
23994                     44    2009-12-21             ATIVO  dez  
23995                     40    2006-02-13             ATIVO  dez  
23996                     30    2016-02-03             ATIVO  dez

## Visão Geral dos Dados

Nesta etapa inicial da Análise Exploratória de Dados (EDA), é apresentada uma visão geral do conjunto de dados que será utilizado nas análises subsequentes.

In [10]:
df_serv.shape

(23997, 12)

- O dataset tem 23997 linhas e 12 colunas

In [12]:
df_serv["id_servidor"].nunique()

980

- Durante o ano de 2025 a Prefeitura Municipal de Santa Rita do Passa Quatro/SP teve 980 servidores.

In [15]:
df_serv["mes"].unique()

array(['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set',
       'out', 'nov', 'dez'], dtype=object)

- A análise cobre todos os meses referentes ao ano de 2025

In [16]:
df_serv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23997 entries, 0 to 23996
Data columns (total 12 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   id_servidor            23997 non-null  object 
 1   sexo                   23997 non-null  object 
 2   cargo                  23997 non-null  object 
 3   categoria_cargo        23997 non-null  object 
 4   tipo_pagamento         23997 non-null  object 
 5   proventos              23919 non-null  float64
 6   descontos              23997 non-null  float64
 7   liquido                23997 non-null  float64
 8   carga_horaria_semanal  23997 non-null  int64  
 9   data_admissao          23997 non-null  object 
 10  data_desligamento      23997 non-null  object 
 11  mes                    23997 non-null  object 
dtypes: float64(3), int64(1), object(8)
memory usage: 2.2+ MB


### Dimensão do dataset
O dataset é composto por **23997 registros** e **12 colunas**, representando pagamentos realizados a servidores públicos ao longo dos meses analisados.

### Servidores únicos
Foram identificados **980 servidores únicos**, considerando o identificador `id_servidor`.  
É importante destacar que um mesmo servidor pode aparecer em múltiplos registros, uma vez que os dados estão organizados por período de pagamento.

### Período de referência
Os dados contemplam os seguintes meses de referência:

Janeiro, Fevereiro, Março, Abril, Maio, Junho, Julho, Agosto, Setembro, Outubro, Novembro e Dezembro

Essa distribuição mensal permite análises temporais sobre a evolução da folha de pagamento ao longo do tempo.

### Estrutura dos dados
O dataset consolidado contém informações relacionadas a:

- Identificação do servidor
- Características funcionais (cargo, categoria e tipo de contrato)
- Informações temporais (data de admissão e desligamento)
- Dados financeiros (proventos, descontos e salário líquido)
- Carga horária semanal

A estrutura atual do dataset está preparada para análises descritivas, comparativas e temporais, que serão exploradas nas próximas etapas da EDA.
